src/pfe/file-watcher/server/test/functional-test/suites/generic/tests/logging.ts

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { expect } from "chai";
+import * as _ from "lodash";
+
+import * as genericLib from "../../../lib/generic";
+import { fail } from "assert";
+
+export default class LoggingTest {
+    testName: string;
+
+    constructor(testName: string) {
+        this.testName = testName;
+    }
+
+    run(runOnly?: boolean): void {
+        (runOnly ? describe.only : describe)(this.testName, () => {
+            const logLevels = ["info", "warn", "error", "trace", "debug", "invalid"];
```

Comment:
It's a general comment, we can define the variable type to make the variable more strict and readable.
eg.
```
const logLevels: Array<string> = ["info", "warn", "error", "trace", "debug", "invalid"];
const invalidLevel: string = _.cloneDeep(logLevels).pop();
```
I find some variables we define some we don't, in order not to block this PR to merge, we can do this improvement when we extend the test cases for other project types

Reply:
Yes, I fixed it.